### PR TITLE
added branch id support

### PR DESCRIPTION
**Pull request by Volodymyr95**

### CODE REVIEW

File: src/main/java/com/crowdin/cli/client/CrowdinProjectClient.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ private void populateProjectWithStructure(CrowdinProjectFull project) {
         project.setFiles(executeRequestFullList((limit, offset) -> this.client.getSourceFilesApi()
             .listFiles(this.projectId, null, null, null, null, limit, offset)));
         project.setDirectories(executeRequestFullList((limit, offset) -> this.client.getSourceFilesApi()
-            .listDirectories(this.projectId, null, null, null, limit, offset)));
+            .listDirectories(this.projectId, null, null, null,null, limit, offset)));
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/client/CrowdinProjectClient.java]:
```suggestion
            .listDirectories(this.projectId, null, null, null, null, limit, offset)));
```

File: src/main/java/com/crowdin/cli/commands/functionality/RequestBuilder.java [hunk 2]
```diff
@@ -160,9 +160,10 @@ public static ExportProjectTranslationRequest exportProjectTranslation(ExportPro
     }
 
     public static CrowdinTranslationCraeteProjectPseudoBuildForm crowdinTranslationCreateProjectPseudoBuildForm(
-        Boolean pseudo, Integer lengthCorrection, String prefix, String suffix, CharTransformation charTransformation
+        Boolean pseudo, Integer lengthCorrection, String prefix, String suffix, CharTransformation charTransformation, long branchId
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/functionality/RequestBuilder.java]: let's the order of the arguments be the same as in the `CrowdinTranslationCraeteProjectPseudoBuildForm` (branch ID first)

also, the `branchId` probably is nullable

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 3]
```diff
@@ -133,8 +133,8 @@ public void act(Outputter out, PropertiesWithFiles pb, ProjectClient client) {
                 PseudoLocalization pl = pb.getPseudoLocalization();
                 BuildProjectTranslationRequest request = (pl != null)
                     ? RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(
-                        true, pl.getLengthCorrection(), pl.getPrefix(), pl.getSuffix(), pl.getCharTransformation())
-                    : RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(true, null, null, null, null);
+                        true, pl.getLengthCorrection(), pl.getPrefix(), pl.getSuffix(), pl.getCharTransformation(), branch.get().getId())
+                    : RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(true, null, null, null, null, 0);
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]: zero branch ID could cause an API error

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 4]
```diff
@@ -133,8 +113,8 @@ public void act(Outputter out, PropertiesWithFiles pb, ProjectClient client) {
                 PseudoLocalization pl = pb.getPseudoLocalization();
                 BuildProjectTranslationRequest request = (pl != null)
                     ? RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(
-                        true, pl.getLengthCorrection(), pl.getPrefix(), pl.getSuffix(), pl.getCharTransformation())
-                    : RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(true, null, null, null, null);
+                    branch.get().getId(),true, pl.getLengthCorrection(), pl.getPrefix(), pl.getSuffix(), pl.getCharTransformation())
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]:
```suggestion
                    branch.get().getId(), true, pl.getLengthCorrection(), pl.getPrefix(), pl.getSuffix(), pl.getCharTransformation())
```

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 5]
```diff
@@ -133,8 +113,8 @@ public void act(Outputter out, PropertiesWithFiles pb, ProjectClient client) {
...
+                    : RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(1L,true, null, null, null, null);
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]:
```suggestion
                    : RequestBuilder.crowdinTranslationCreateProjectPseudoBuildForm(1L, true, null, null, null, null);
```

File: src/main/java/com/crowdin/cli/commands/functionality/RequestBuilder.java [hunk 6]
```diff
@@ -159,10 +159,11 @@ public static ExportProjectTranslationRequest exportProjectTranslation(ExportPro
         return copy;
     }
 
-    public static CrowdinTranslationCraeteProjectPseudoBuildForm crowdinTranslationCreateProjectPseudoBuildForm(
-        Boolean pseudo, Integer lengthCorrection, String prefix, String suffix, CharTransformation charTransformation
+    public static CrowdinTranslationCraeteProjectPseudoBuildForm crowdinTranslationCreateProjectPseudoBuildForm(long branchId,
+                                                                                                                Boolean pseudo, Integer lengthCorrection, String prefix, String suffix, CharTransformation charTransformation
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/functionality/RequestBuilder.java]: ???

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 7]
```diff
@@ -32,28 +27,13 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
-import java.util.ArrayList;
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]: Let's not change these imports

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java [hunk 8]
```diff
@@ -32,28 +27,13 @@
...
 import java.util.stream.Collectors;
 
-import static com.crowdin.cli.BaseCli.CHECK_WAITING_TIME_FIRST;
-import static com.crowdin.cli.BaseCli.CHECK_WAITING_TIME_INCREMENT;
-import static com.crowdin.cli.BaseCli.CHECK_WAITING_TIME_MAX;
-import static com.crowdin.cli.BaseCli.RESOURCE_BUNDLE;
-import static com.crowdin.cli.utils.console.ExecutionStatus.ERROR;
-import static com.crowdin.cli.utils.console.ExecutionStatus.OK;
-import static com.crowdin.cli.utils.console.ExecutionStatus.WARNING;
+import static com.crowdin.cli.BaseCli.*;
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]: and these imports also

File: src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java (hunk 5 above)
Comment (Volodymyr95) [reply src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]: don't understand this comment

Comment (andrii-bodnar) [reply src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java]: This is a code suggestion just to add a space after the first parameter (code formatting)

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**